File: backend/src/main/java/com/exp_planner/backend/domain/planner/dto/ExpenseDto.java
package com.exp_planner.backend.domain.planner.dto;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE) // 객체로 만들 필요 없으니 private
public class ExpenseDto {

    @Getter
    @NoArgsConstructor
    public static class Post {
        private String gameName;
        private String gameType;
        private Integer amount;
        private LocalDate paymentDate;
        private String paymentMethod;
        private String marketType;
        private String description;
    }

    @Getter
    @NoArgsConstructor
    public static class Response {
        private Long id;
        private String gameName;
        private String gameType;
        private Integer amount;
        private LocalDate paymentDate;
        private String paymentMethod;
        private String marketType;
        private String description;
    }
}

File: backend/src/main/java/com/exp_planner/backend/domain/planner/dto/ExpenseDto.kt
package com.exp_planner.backend.domain.planner.dto

import java.time.LocalDate

class ExpenseDto {
    data class Post(
        val gameName: String,
        val gameType: String,
        val amount: Int,
        val paymentDate: LocalDate,
        val paymentMethod: String?, // null 가능 필드는 ?
        val marketType: String,
        val description: String?
    )

    data class Response(
        val id: Long,
        val gameName: String,
        val gameType: String,
        val amount: Int,
        val paymentDate: LocalDate,
        val paymentMethod: String?,
        val marketType: String,
        val description: String?
    )
}